core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -1370,4 +1370,12 @@ default boolean isSupportedVersionType(ConnectorSession session, SchemaTableName
     {
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support versioned tables");
     }
+
+    /**
+     * Checks if a table exists.
+     */
+    default boolean tableExists(ConnectorSession session, SchemaTableName tableName)
+    {
+        return listTables(session, Optional.of(tableName.getSchemaName())).contains(tableName);
```

Comment:
This is a quick implementation that indeed works as intended.
However this implementation is quite wasteful in terms of resources. We don't need to retrieve all the tables from the database/metastore, but instead we want to verify if only a specified table does or does not exist.
This means that we'll have to get our hands a bit dirty and implement this method across all the connectors. 

Benefit of doing this - getting a better understanding of each of the connectors.